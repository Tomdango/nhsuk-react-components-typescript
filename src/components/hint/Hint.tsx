import React, { HTMLProps } from 'react';
import classNames from 'classnames';

const Hint: React.FC<HTMLProps<HTMLSpanElement>> = ({ className, ...rest }) => (
  <span className={classNames('nhsuk-hint', className)} {...rest}></span>
);

export default Hint;
