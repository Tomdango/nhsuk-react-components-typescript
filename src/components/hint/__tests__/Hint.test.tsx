import React from 'react';
import Hint from '..';
import { shallow } from 'enzyme';

describe('Hint', () => {
  it('matches snapshot', () => {
    const component = shallow(<Hint>Text</Hint>);
    expect(component.text()).toBe('Text');
    expect(component).toMatchSnapshot('Hint');
    component.unmount();
  });
});
