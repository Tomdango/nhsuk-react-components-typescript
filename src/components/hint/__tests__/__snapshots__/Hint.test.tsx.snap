// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint matches snapshot: Hint 1`] = `
<span
  className="nhsuk-hint"
>
  Text
</span>
`;
